packages/core/src/storages/sitemap_request_list.ts

```diff
@@ -165,14 +234,28 @@ export class SitemapRequestList implements IRequestList {
         this.sitemapParsingProgress.pendingSitemapUrls = new Set(options.sitemapUrls);
     }
 
+    private isUrlMatchingPatterns(url: string): boolean {
+        return (
+            !this.urlExcludePatternObjects.some((patternObject) => {
+                const { regexp, glob } = patternObject;
+                return (regexp && url.match(regexp)) || (glob && minimatch(url, glob, { nocase: true }));
+            }) &&
+            (this.urlPatternObjects.length === 0 ||
+                this.urlPatternObjects.some((patternObject) => {
+                    const { regexp, glob } = patternObject;
+                    return (regexp && url.match(regexp)) || (glob && minimatch(url, glob, { nocase: true }));
+                }))
```

Comment:
Bruh, extract those massive boolean expressions into variables please.

Reply:
I first thought of a function that would return these arrow functions, closuring a URL (so we could only write `this.urlPatternObjects.some(matchUrl(url))` - imo beautifully readable). Both of the arrow functions currently are the same thing. I just didn't want to define this `matchUrl` function in the middle of `isUrlMatchingPatterns` (as it's gonna be called many many many times) - and making it a private method in `SitemapRequestList` didn't really feel right either.

Idk, what do you think?